app/models/manageiq/providers/cisco_intersight/inventory/parser/physical_infra_manager.rb

```diff
@@ -3,200 +3,472 @@ class Inventory::Parser::PhysicalInfraManager < Inventory::Parser
+        :location           => format_location(device_contract_information_unit),
+        :location_led_state => get_locator_led_state(chassis),
+        :model              => chassis.model,
+        :part_number        => chassis.part_number,
+        :resource           => physical_chassis,
+        :serial_number      => chassis.serial,
+        :product_name       => get_product_name(device_contract_information_unit),
+        :machine_type       => get_machine_type(device_contract_information_unit)
+      )
+    end
+
+    def build_physical_racks(rack)
+      # Was not able to test it yet due to not having physical racks written on the intersight side with the current lab rights
+      # TODO: Find out which API call should be used here (after racks are viewable on the API viewer).
+      # Builds out collection physical_racks
+      # Object types:
+      #   - rack - <I don't know yet>, object obtained by intersight client
+      # Returns:
+      #   ManageIQ's object ManageIQ::Providers::CiscoIntersight::PhysicalInfraManager::PhysicalChassis
+
+      persister.physical_racks.build(
+        :ems_ref => rack.moid,
+        :name    => ""
+      )
+    end
+
+    def build_physical_servers(server)
+      # Builds out collection physical_servers
+      # Object types:
+      #   - server - ComputePhysicalSummary, object obtained by intersight client
+      #   - physical_rack - ManageIQ::Providers::CiscoIntersight::PhysicalInfraManager::PhysicalRacks
+      #   - physical_chassis - ManageIQ::Providers::CiscoIntersight::PhysicalInfraManager::PhysicalChassis
+      # Returns:
+      #   ManageIQ's object ManageIQ::Providers::CiscoIntersight::PhysicalInfraManager::PhysicalServers
+
+      # Obtain data about chassis and racks if they're wired to the server
+      physical_rack = nil # TODO: After chassis gets written into the DB, obtain it and write it here as reference using lazy find.
+      physical_chassis = persister.physical_chassis.lazy_find(server.equipment_chassis.moid) if server.equipment_chassis
+      registered_device_moid = get_registered_device_moid(server)
+      device_registration = collector.get_asset_device_registration_by_moid(registered_device_moid)
+      persister.physical_servers.build(
+        :ems_ref          => server.moid,
+        :health_state     => get_health_state(server), # health_state obtained through atribute s.alarm_summary
+        :hostname         => device_registration.device_hostname[0], # I assume one registered device manages one (and only one) compute element
+        :name             => server.name,
+        :physical_chassis => physical_chassis,
+        :physical_rack    => physical_rack, # nil for now
+        :power_state      => server.admin_power_state,
+        :raw_power_state  => server.admin_power_state,
+        :manufacturer     => server.vendor,
+        :type             => "ManageIQ::Providers::CiscoIntersight::PhysicalInfraManager::PhysicalServer"
```

Comment:
This _should_ be automatically built by the core persister builder class, try leaving this out and add a spec testing that it is set properly